src/main/scala/firrtl/transforms/DedupAnnotations.scala

```diff
@@ -61,7 +61,7 @@ object DedupAnnotationsTransform {
     }
 
     // Partition the dedupable annotations into groups that *should* deduplicate into the same annotation
-    val shouldDedup: Map[Any, ArrayBuffer[DedupableRepr]] = canDedup.groupBy(_.dedupKey)
+    val shouldDedup: Seq[(Any, Seq[DedupableRepr])] = Utils.groupByIntoSeq(canDedup.toSeq)(_.dedupKey)
```

Comment:
```suggestion
    val shouldDedup: Seq[(Any, Seq[DedupableRepr])] = groupByIntoSeq(canDedup)(_.dedupKey)
```
Rather than casting to a `Seq`, I think it'd be better to just make `groupByIntoSeq` accept an iterable.
Also no need for `Utils.` scoping if you're importing.